daemon/mgr/image.go

```diff
@@ -283,6 +288,84 @@ func (mgr *ImageManager) AddTag(ctx context.Context, sourceImage string, targetT
 	return err
 }
 
+// ImageHistory returns image history by reference.
+func (mgr *ImageManager) ImageHistory(ctx context.Context, idOrRef string) ([]types.HistoryResultItem, error) {
+	img, err := mgr.fetchContainerdImage(ctx, idOrRef)
+	if err != nil {
+		return nil, err
+	}
+
+	desc, err := img.Config(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	ociImage, err := containerdImageToOciImage(ctx, img)
+	if err != nil {
+		return nil, err
+	}
+
+	// layers info
+	cs := img.ContentStore()
+	manifest, err := ctrdmetaimages.Manifest(ctx, cs, img.Target(), platforms.Default())
```

Comment:
could we extract the function for this part? call the `getLayer`, how do you think?